src/particle.f90

```diff
@@ -0,0 +1,2187 @@
+!Copyright (c) 2012-2022, Xcompact3d
+!This file is part of Xcompact3d (xcompact3d.com)
+!SPDX-License-Identifier: BSD 3-Clause
+module particle
+
+  use decomp_2d_constants, only : mytype
+  use decomp_2d_mpi, only : nrank,nproc,decomp_2d_abort
+  use mptool
+  use constants, only : pi
+
+  implicit none
+
+  interface mclean
+    module procedure mclean_mytype
+    module procedure mclean_particles
+  end interface mclean 
+  !
+  interface msize
+    module procedure size_particle
+    module procedure size_integer
+  end interface msize
+  !
+  interface pa2a
+    module procedure pa2a_particle
+  end interface pa2a
+  !
+  interface mextend
+     module procedure extend_particles
+  end interface mextend
+  !
+  ! particles
+  type partype
+
+    private
```

Comment:
it is designed that the particle type is only accessable within the module. the external routines will have to use subrotines inside the particle module to access particle.

Reply:
Having the `partype` type private and hidden from the rest of the code forces the particle development to stay inside a single module / file. On one hand, it seems easier because everything is located at a single location. On the other hand, it forces all the particle-related stuff to be located here, which can lead to issues. It might be relevant to split the particle module into several modules to allow a modular approach. For instance : `particle_kernel` could contain core and computing part, `particle_io` all the IO for particles, `particle_tools` utilities for particles, `particle_comm` MPI communications specific to particles, `particle_postproc` post-processing tools specific to particles, ...

If the particle module is supposed to grow, the modular approach is probably much more convenient.

Reply:
Good idea! I will have a try to see if it is too much to do..

Reply:
If it is too much work, keep it for later

Reply:
Pushed the change by spliiting the module into 3, but still keep everything in a single file.